### DISCUSSION TITLE
Unable to extract "Oblique font style" and "Helvetica font family" being extracted in two different ways in two different machines

### DISCUSSION THREAD

**Discussion by SirishaGorasa** [pymupdf/PyMuPDF#1878] [Q&A] [answered]
I am trying to extract font styles and family using the below code:
```
import fitz
document=fitz.open("Styles.pdf")
for p in document:
    for block in p.get_text("dict")["blocks"]:
        if block['type']==0:
            if 'lines' in block.keys():
                for line in block["lines"]:
                    for span in line["spans"]:
                        print(span["font"])
```
I am attaching my input pdf below which has text written in Times New Roman font family in normal, italic and oblique font styles (1st, 2nd and 3rd paragraphs respectively) and also text (4th paragraph) written in Helvetica Font family.
[Styles.pdf](https://github.com/pymupdf/PyMuPDF/files/9359247/Styles.pdf)

The output of the above code is 
```
TimesNewRomanPSMT
TimesNewRomanPS-ItalicMT
TimesNewRomanPS-ItalicMT
Helvetica
Calibri
```
in one machine and 
```
TimesNewRomanPSMT
TimesNewRomanPS-ItalicMT
TimesNewRomanPS-ItalicMT
ArialMT
Calibri
```
in another machine
Oblique font style is also extracted as Italic font style.
In some cases, Helvetica got extracted as ArialMT. Why did this happen? 
Why did Calibri come as the last font in spite of its absence in the input PDF?
Please let me know at the earliest.

**Comment by JorjMcKie** [comment, marked as answer]
This is what I get:
```python
for b in page.get_text("dict",flags=fitz.TEXTFLAGS_TEXT)["blocks"]:
    for l in b["lines"]:
        for s in l["spans"]:
            print(f"font {s['font']}, text: '{s['text']}'")

            
font TimesNewRomanPSMT, text: 'This is a paragraph in normal style. '
font TimesNewRomanPS-ItalicMT, text: 'This is a paragraph in italic style. '
font TimesNewRomanPS-ItalicMT, text: 'This is a paragraph in oblique style. '
font Helvetica, text: 'This is a paragraph in Helvetica font family. '
font Calibri, text: ' '
```
Who wrote 1 space char in Calibri?
Blame him!

**Reply by SirishaGorasa** [reply]
Can't we differentiate between Italic and Oblique font styles?  and 
Why is Helvetica extracted as Arial in one of my machines? Is it dependent on Word version?

**Reply by JorjMcKie** [reply]
Did you create the PDF with word?
Then yes, word does all sorts of things unexpectedly.
If you have the time compare Word's output with that of LibreOffice for example - something very different.

> Can't we differentiate between Italic and Oblique font styles?

No. You need to be aware, that MuPDF does not look at the visual appearance to make its judgement.
It actually doesn't investigate any of this. It just asks the font "Are you italic?" and must accept whatever answer.
If you are able load the font yourself via `fitz.Font(...)` you can look at font.info(). It may show you if italic is simulated ( = oblique).
